python/mxnet/kvstore/kvstore.py

```diff
@@ -349,6 +363,7 @@ def pushpull(self, key, value, out=None, priority=0):
 
         out: NDArray or list of NDArray or list of list of NDArray
             Values corresponding to the keys.
+            `out` should have length as `value`.
```

Comment:
Why is there such a restriction?

Reply:
there is grouping of keys and corresponding ndarrays happening at the backend based on their positions in the list before performing kv operations. Since we are using same key list for both value and out, their layouts should be identical. length check is just for a sanity.

this is the reason why I had both vkeys and okeys separate in the previous implementation.

Reply:
You're right. However, if there's one key, having value = a list of 4 arrays, and out = a list of one array is also valid.

Reply:
in that case we need to bring back separate key list for value and out.